NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -123,136 +636,382 @@ def calculate_simulation_weights(positions, zenith, azimuth, site='summit', debu
-    sim_shower.set_parameter(shp.distance_shower_maximum_geometric,
-                             corsika['CoREAS'].attrs["DistanceOfShowerMaximum"] * units.cm)
-
-    sim_shower.set_parameter(shp.observation_level, corsika["inputs"].attrs["OBSLEV"] * units.cm)
-    sim_shower.set_parameter(shp.primary_particle, corsika["inputs"].attrs["PRMPAR"])
-    if 'ATMOD' in corsika['inputs'].attrs.keys():
-        sim_shower.set_parameter(shp.atmospheric_model, corsika["inputs"].attrs["ATMOD"])
-
-    try:
-        sim_shower.set_parameter(shp.electromagnetic_energy, corsika["highlevel"].attrs["Eem"] * units.eV)
-    except:
-        global warning_printed_coreas_py
-        if(not warning_printed_coreas_py):
-            logger.warning("No high-level quantities in HDF5 file, not setting EM energy, this warning will be only printed once")
-            warning_printed_coreas_py = True
-
-    return sim_shower
+        if debug:
+            max_efield = []
+            for i in range(len(self.electric_field_on_sky[:, 0, 1])):
+                max_efield.append(np.max(np.abs(self.electric_field_on_sky[i, :, :])))
+            plt.scatter(self.obs_positions_vxB_vxvxB[:, 0], self.obs_positions_vxB_vxvxB[:, 1], c=max_efield,
+                        cmap='viridis', marker='o', edgecolors='k')
+            cbar = plt.colorbar()
+            cbar.set_label('max amplitude')
+            plt.xlabel('v x B [m]')
+            plt.ylabel('v x v x B [m]')
+            plt.show()
+            plt.close()
+
+        return self.fluence_interpolator
+
+    def plot_footprint_fluence(self, radius=300):
+        """
+        plots the interpolated values of the fluence in the shower plane
+
+        Parameters
+        ----------
+        radius : float
+            radius around shower core which should be plotted
+
+        Returns
+        -------
+        fig : figure object
+
+        ax : axis object
+        """
+
+        # Make color plot of f(x, y), using a meshgrid
+        ti = np.linspace(-radius, radius, 500)
+        XI, YI = np.meshgrid(ti, ti)
+
+        # Get interpolated values at each grid point, calling the instance of interp2d_fourier
+        ZI = self.fluence_interpolator(XI, YI)
+
+        # And plot it
+        maxp = np.max(ZI)
+        fig, ax = plt.subplots(figsize=(8, 6))
+        ax.pcolor(XI, YI, ZI, vmax=maxp, vmin=0, cmap=cm.gnuplot2_r)
+        mm = cm.ScalarMappable(cmap=cm.gnuplot2_r)
+        mm.set_array([0.0, maxp])
+        cbar = plt.colorbar(mm, ax=ax)
+        cbar.set_label(r'energy fluence [eV/m^2]', fontsize=14)
+        ax.set_xlabel(r'$\vec{v} \times \vec{B} [m]', fontsize=16)
+        ax.set_ylabel(r'$\vec{v} \times (\vec{v} \times \vec{B})$ [m]', fontsize=16)
+        ax.set_xlim(-radius, radius)
+        ax.set_ylim(-radius, radius)
+        ax.set_aspect('equal')
+        return fig, ax
+
+    def get_interp_efield_value(self, position_on_ground, core):
+        """
```

Comment:
I am removing the core parameter, because I think it is very confusing. This module is purely about the interpolation,so everything should be from the reference frame of the original starshape IMHO. If one wants to shift the antenna positions, to account for a core shift, I think the should do this higher up and subtract the core from the `position_on_ground` value. What do you think?

Reply:
sure, we can move the function to another place. I placed it here because the interpolator object is stored in that module. Ideally, it would live in readCoREASDetector.py I guess.